src/components/app/App.test.tsx

```diff
@@ -0,0 +1,290 @@
+import { httpGet } from '../../utils/httpGet';
+import App from './App';
+import { PAGE_NUMBER_DEFAULT } from '../../utils/constants';
+import type { ISectionsResponseResults, ISectionsResponse, ISearchResponseResults, ISearchResponse } from '../../types/types';
+
+describe(`App`, () => {
+    beforeAll(() => {
+        window.scrollTo = jest.fn();
+    })
+
+    beforeEach(() => {
+        jest.useFakeTimers()
+    });
+
+    afterEach(() => {
+        jest.runOnlyPendingTimers()
+        jest.useRealTimers()
+    });
+
+    const mockedSectionsResponseResultsList: ISectionsResponseResults[] = Array.from(new Array(20).fill(null).map((_, idx) => (
+        {
+            id: `test-section-id-${idx + 1}`,
+            webTitle: `test-section-webTitle-${idx + 1}`,
+            webUrl: 'test-section-webUrl',
+        }
+    )));
+
+    const mockedSectionsResponse: ISectionsResponse = {
+        results: mockedSectionsResponseResultsList,
+    };
+
+    const mockedSearchResponseResultsList: ISearchResponseResults[] = Array.from(new Array(10).fill(null).map((_, idx) => (
+        {
+            id: `test-article-id-${idx + 1}`,
+            type: 'test-article-type',
+            sectionId: 'test-article-sectionId',
+            sectionName: 'test-article-sectionName',
+            webPublicationDate: new Date('2023-01-01').toString(),
+            webTitle: 'test-article-webTitle',
+            webUrl: 'test-article-webUrl',
+        }
+    )));
+
+    const mockedSearchResponse: ISearchResponse = {
+        total: 101,
+        startIndex: 1,
+        pageSize: 10,
+        currentPage: 1,
+        pages: 11,
+        results: mockedSearchResponseResultsList,
+    };
+
+    describe(`requests failed`, () => {
+        it(`should render "sections" and "search" loading error indication`, async () => {
+            jest.spyOn(httpGetModule, "httpGet").mockImplementation(() => {
+                return Promise.reject(new Error('test-error'));
+            });
+
+            await act(async () => {
+                render(
+                    <MemoryRouter initialEntries={['']}>
+                        <App />
+                    </MemoryRouter>
+                );
+            });
+
+            expect(httpGet).toHaveBeenCalledTimes(2);
+            expect(httpGet).toHaveBeenNthCalledWith(1, 'sections');
+            expect(httpGet).toHaveBeenNthCalledWith(2, 'search', {});
+            expect(httpGet).rejects.toThrowError(new Error('test-error'));
+            expect(httpGet).rejects.not.toThrowError(new Error('test-errorXYZ'));
+            expect(screen.getAllByText(/Oops! Something went wrong/i)).toHaveLength(2);
+        });
+    });
+
+    describe(`requests pending`, () => {
+        it(`should render App component with "sections" and "articles" loading indication`, async () => {
+            jest.spyOn(httpGetModule, "httpGet").mockImplementation((url: string, params?: Record<string, string>) => {
```

Comment:
You have a lot small irrelevant issues like sometimes using double quotes, missing semicolons etc. Lets talk about adding linter to the project to easily find all those issues for you.